src/App.tsx

```diff
@@ -15,49 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+  Reset = 'reset',
+}
+
 export const App: React.FC = () => {
+  const [viewGoods, setViewGoods] = useState(goodsFromServer);
+  const [sortType, setSortType] = useState<SortType | null>(null);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const isSorted = !viewGoods.every(
+    (value, index) => value === goodsFromServer[index],
+  );
```

Comment:
The `isSorted` logic checks if the `viewGoods` array matches the `goodsFromServer` array in order. This may not work as expected if the list is reversed. Consider using a more robust method to check if the list has been altered, such as comparing the arrays directly or using a flag to track changes.